src/mpl2/src/object.cpp

```diff
@@ -1106,6 +1124,19 @@ void HardMacro::getDirectionTrackGrid(odb::dbTrackGrid* track_grid,
   }
 }
 
+// Offset between the terminal and the adjacent line of the track-grid.
+float HardMacro::computeMTermOffset(const float pin_to_origin,
+                                    const float pitch)
+{
+  float distance_from_origin = 0.0f;
+
+  while (distance_from_origin <= pin_to_origin) {
+    distance_from_origin += pitch;
+  }
+
+  return pin_to_origin - (distance_from_origin - pitch);
```

Comment:
Indeed. Thanks. I didn't notice that it could be done this way.

Within this context, I don't think we need this new function. So, regarding the other comment about computing in DBU, would we need to be concerned about computing the offset in micron?

Reply:
In general it is a [bad idea](https://github.com/The-OpenROAD-Project/OpenROAD/blob/master/docs/contrib/DatabaseMath.md) to do things in microns (though it happens a lot in mpl2).  In this case I see you are doing
```
pin_to_origin_x = dbuToMicron(mpin->getBBox().xMin(), dbu_);
```
before calling this method.  Can we do the conversion after (or not all)?

Reply:
I think we can. I'll take a look more carefully.